transformers/component_tests/queries/get_flap_query_test.go

```diff
@@ -77,15 +81,15 @@ var _ = Describe("Get flap query", func() {
 		Expect(err).NotTo(HaveOccurred())
 
 		flapStorageValuesTwo := test_helpers.GetFlapStorageValues(2, fakeBidId)
-		test_helpers.CreateFlap(db, headerTwo, flapStorageValuesTwo, test_helpers.GetFlapMetadatas(strconv.Itoa(fakeBidId)), contractAddress)
+		test_helpers.CreateFlap(db, diffID, headerTwo.Id, flapStorageValuesTwo, test_helpers.GetFlapMetadatas(strconv.Itoa(fakeBidId)), contractAddress)
 
 		flapStorageValuesOne := test_helpers.GetFlapStorageValues(1, fakeBidId)
-		test_helpers.CreateFlap(db, headerOne, flapStorageValuesOne, test_helpers.GetFlapMetadatas(strconv.Itoa(fakeBidId)), contractAddress)
+		test_helpers.CreateFlap(db, diffID, headerOne.Id, flapStorageValuesOne, test_helpers.GetFlapMetadatas(strconv.Itoa(fakeBidId)), contractAddress)
 
 		// creating another block + updated storage values to ensure that get_flap uses the specified block
 		headerThree := createHeader(blockTwo+1, timestampTwo+1, headerRepo)
 		flapStorageValuesThree := test_helpers.GetFlapStorageValues(3, fakeBidId)
-		test_helpers.CreateFlap(db, headerThree, flapStorageValuesThree, test_helpers.GetFlapMetadatas(strconv.Itoa(fakeBidId)), contractAddress)
+		test_helpers.CreateFlap(db, diffID, headerThree.Id, flapStorageValuesThree, test_helpers.GetFlapMetadatas(strconv.Itoa(fakeBidId)), contractAddress)
```

Comment:
Might be overthinking this, but I'd be interested in not reusing the same diff ID for multiple inserts - especially when dealing with different headers. Haven't gotten to the test_helpers yet, but I wonder if those functions could create diffs on the fly for each `repo.Create` call

Reply:
nice catch! updated in f9a8617